### PR TITLE
HBASE-22365 Region may be opened on two RegionServers

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache9** [comment]
@infraio I think we can try this in our internal branch first to see if ITBLL could pass?

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 303 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 327 | master passed |
| +1 | compile | 70 | master passed |
| +1 | checkstyle | 92 | master passed |
| +1 | shadedjars | 352 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 293 | master passed |
| +1 | javadoc | 42 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 326 | the patch passed |
| +1 | compile | 70 | the patch passed |
| +1 | javac | 70 | the patch passed |
| +1 | checkstyle | 91 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 348 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 642 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 266 | the patch passed |
| +1 | javadoc | 41 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 15450 | hbase-server in the patch failed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 18823 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestAdmin1 |
|   | hadoop.hbase.TestMetaTableLocator |
|   | hadoop.hbase.filter.TestFilterListOnMini |
|   | hadoop.hbase.util.TestFSUtils |
|   | hadoop.hbase.TestMultiVersions |
|   | hadoop.hbase.TestAcidGuaranteesWithBasicPolicy |
|   | hadoop.hbase.util.TestFromClientSide3WoUnsafe |
|   | hadoop.hbase.regionserver.wal.TestLogRollAbort |
|   | hadoop.hbase.TestClientClusterStatus |
|   | hadoop.hbase.regionserver.TestClusterId |
|   | hadoop.hbase.TestMetaTableAccessor |
|   | hadoop.hbase.TestLocalHBaseCluster |
|   | hadoop.hbase.TestRegionMetrics |
|   | hadoop.hbase.regionserver.TestScannerWithCorruptHFile |
|   | hadoop.hbase.TestRegionLoad |
|   | hadoop.hbase.TestHBaseTestingUtility |
|   | hadoop.hbase.regionserver.wal.TestAsyncWALReplay |
|   | hadoop.hbase.regionserver.TestCleanupMetaWAL |
|   | hadoop.hbase.zookeeper.TestZooKeeperACL |
|   | hadoop.hbase.filter.TestFuzzyRowAndColumnRangeFilter |
|   | hadoop.hbase.filter.TestFuzzyRowFilterEndToEnd |
|   | hadoop.hbase.regionserver.wal.TestSecureWALReplay |
|   | hadoop.hbase.util.TestHBaseFsckEncryption |
|   | hadoop.hbase.TestSequenceIdMonotonicallyIncreasing |
|   | hadoop.hbase.util.TestHBaseFsckReplication |
|   | hadoop.hbase.filter.TestScanRowPrefix |
|   | hadoop.hbase.TestInfoServers |
|   | hadoop.hbase.regionserver.TestEncryptionRandomKeying |
|   | hadoop.hbase.TestJMXListener |
|   | hadoop.hbase.util.TestHBaseFsckMOB |
|   | hadoop.hbase.io.encoding.TestChangingEncoding |
|   | hadoop.hbase.regionserver.wal.TestLogRolling |
|   | hadoop.hbase.io.encoding.TestEncodedSeekers |
|   | hadoop.hbase.regionserver.TestMutateRowsRecovery |
|   | hadoop.hbase.procedure.TestFailedProcCleanup |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/221 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d1112bc53165 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 45bcb4f307 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/2/testReport/ |
| Max. process+thread count | 4847 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 266 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 256 | master passed |
| +1 | compile | 56 | master passed |
| +1 | checkstyle | 71 | master passed |
| +1 | shadedjars | 257 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 212 | master passed |
| +1 | javadoc | 30 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 242 | the patch passed |
| +1 | compile | 52 | the patch passed |
| +1 | javac | 52 | the patch passed |
| +1 | checkstyle | 68 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 265 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 487 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 196 | the patch passed |
| +1 | javadoc | 34 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 16720 | hbase-server in the patch failed. |
| +1 | asflicense | 22 | The patch does not generate ASF License warnings. |
| | | 19311 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.client.TestFromClientSide |
|   | hadoop.hbase.master.TestAssignmentManagerMetrics |
|   | hadoop.hbase.regionserver.TestEncryptionKeyRotation |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectoryWithRegionReplicas |
|   | hadoop.hbase.client.TestSnapshotTemporaryDirectory |
|   | hadoop.hbase.client.TestFromClientSide3 |
|   | hadoop.hbase.client.TestFromClientSideWithCoprocessor |
|   | hadoop.hbase.client.TestSnapshotDFSTemporaryDirectory |
|   | hadoop.hbase.client.TestAdmin1 |
|   | hadoop.hbase.replication.TestReplicationDroppedTables |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/221 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 509b4284d12f 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 45bcb4f307 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/1/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/1/testReport/ |
| Max. process+thread count | 5508 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Oh, let me check the failed UTs.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 21 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 240 | master passed |
| +1 | compile | 52 | master passed |
| +1 | checkstyle | 69 | master passed |
| +1 | shadedjars | 258 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 189 | master passed |
| +1 | javadoc | 33 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 235 | the patch passed |
| +1 | compile | 52 | the patch passed |
| +1 | javac | 52 | the patch passed |
| +1 | checkstyle | 65 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 259 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 483 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 194 | the patch passed |
| +1 | javadoc | 32 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8264 | hbase-server in the patch passed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 10555 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/221 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux d7ae4bda64aa 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 67c937fa92 |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/3/testReport/ |
| Max. process+thread count | 4704 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionRemoteProcedureBase.java [hunk 1]
```diff
@@ -175,7 +176,7 @@ void reportTransition(MasterProcedureEnv env, RegionStateNode regionNode, Server
       throw new UnexpectedStateException("Received report from " + serverName + ", expected " +
         targetServer + ", " + regionNode + ", proc=" + this);
     }
-    reportTransition(regionNode, transitionCode, seqId);
+    reportTransition(env, regionNode, transitionCode, seqId);
     // this state means we have received the report from RS, does not mean the result is fine, as we
```
[the remark this answers is not in the document]
Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionRemoteProcedureBase.java]: Yes, fixed. Please see the updated patch. I introduced a checkTransition and then an updateTransition method.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ master Compile Tests _ |
| +1 | mvninstall | 267 | master passed |
| +1 | compile | 53 | master passed |
| +1 | checkstyle | 70 | master passed |
| +1 | shadedjars | 260 | branch has no errors when building our shaded downstream artifacts. |
| +1 | findbugs | 187 | master passed |
| +1 | javadoc | 31 | master passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 231 | the patch passed |
| +1 | compile | 51 | the patch passed |
| +1 | javac | 51 | the patch passed |
| +1 | checkstyle | 70 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 260 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 487 | Patch does not cause any errors with Hadoop 2.7.4 or 3.0.0. |
| +1 | findbugs | 193 | the patch passed |
| +1 | javadoc | 31 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 8146 | hbase-server in the patch passed. |
| +1 | asflicense | 23 | The patch does not generate ASF License warnings. |
| | | 10474 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/221 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux ac791fc9d548 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | master / 420fbba6ae |
| maven | version: Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T18:33:14Z) |
| Default Java | 1.8.0_181 |
| findbugs | v3.1.11 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/4/testReport/ |
| Max. process+thread count | 4996 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-221/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**